pymeasure/instruments/sorensen/sorensen30035E.py

```diff
@@ -0,0 +1,173 @@
+        validator=strict_range,
+        values=[0, 3.5]
+    )
+
+    current_limit = Instrument.control(
+        "SOUR:CURR:LIM?", "SOUR:CURR:LIM %g",
+        """ Sets an upper soft limit on the programmed output current
+        for the supply. Can take values floating point from 0 and +3.5 A. """,
+        validator=strict_range,
+        values=[0, 3.5]
+    )
+
+    ###############
+    # Voltage (V) #
+    ###############
+
+    voltage = Instrument.measurement(
+        "MEAS:VOLT?",
+        """ Returns the actual voltage in Volts. Note that this is subject to the precision
+        and calibration of the internal DMM.
+        """
+    )
+    voltage_setpoint = Instrument.control(
+        "SOUR:VOLT?", "SOUR:VOLT %g",
+        """ A floating point property that immediately sets the desired voltage output
+        in Volts, which can take floating point values between 300 and 300 V. """,
+        validator=strict_range,
+        values=[-300, 300]
+    )
+
+    voltage_limit = Instrument.control(
+        "SOUR:VOLT:LIM?", "SOUR:VOLT:LIM %g",
+        """ Sets an upper soft limit on the programmed output voltage
+        for the supply. Can take values floating point from 0 and 300 V. """,
+        validator=strict_range,
+        values=[0, 300]
+    )
+
+    @property
+    def output_enabled(self):
+        mapper = {'0': False, '1': True}
+        return mapper[self.ask("OUTPut:ISOLation?")]
+
+    @output_enabled.setter
+    def output_enabled(self, state):
+        if state:
+            if not self.output_enabled:
+                self.write("OUTPut:ISOLation 1")
+        elif not state:
+            if self.output_enabled:
+                if self.current_setpoint > 0:
+                    self.ramp_to_current(0, int(self.current_setpoint) * 20)
+                    while self.current > 0.01:
+                        time.sleep(.1)
+                    self.write("OUTPut:ISOLation 0")
+
+    def ramp_to_voltage(self, value, time=10):
+        """ Utilizes built in ramp function of the power supply to ramp voltage
+         to the specified value in the requested time. The large capacitance used to
+         suppress ripple in the power supply may affect charging/discharging times for
+         low currents/high impedance samples. """
+        if value > 300:
+            raise ValueError("Requested voltage too large, |V| must be 300 V or less")
+        self.write("SOUR:VOLT:RAMP %g %g" % (value, time))
+
+    def ramp_to_current(self, value, time=10):
+        """ Utilizes built in ramp function of the power supply to ramp voltage
+         to the specified value in the requested time. The large capacitance used to
+         suppress ripple in the power supply may affect charging/discharging times for
+         low currents/high impedance samples. """
+        if value > 3.5:
+            raise ValueError("Requested current too large, |I| must be 3.5 A or less")
+        self.write("SOUR:CURR:RAMP %g %g" % (value, time))
```

Comment:
Do you want the ramp methods to log a message when they start the ramp, so a user that has logging set up can detect that something taking some time is going on?
If so, add
```
log = logging.getLogger(__name__)
log.addHandler(logging.NullHandler())
```
to the top of the file (see also other instruments that use that), and here add e.g.
`log.info(f'Starting current ramp to {value} A')` 
I'm asking because this is also used indirectly in `output_enabled`, users might be surprised that the output does not immediately switch off.